orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationConfig.java

```diff
@@ -0,0 +1,136 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.sap.ai.sdk.core.AiModel;
+import io.vavr.control.Option;
+import javax.annotation.Nonnull;
+
+/**
+ * Represents the configuration for the orchestration service. Allows for configuring the different
+ * modules of the orchestration service via a fluent API.
+ *
+ * <p>The orchestration pipeline combines different modules into a single execution flow where the
+ * output of one module serves as input for the next. The pipeline consists of the following
+ * modules:
+ *
+ * <ul>
+ *   <li>LLM Config (Mandatory)
+ *   <li>Templating (Optional)
+ *   <li>Data Masking (Optional)
+ *   <li>Input Content Filtering (Optional)
+ *   <li>Output Content Filtering (Optional)
+ * </ul>
+ *
+ * @param <T> Type of the specific implementation to make a fluent API possible.
+ */
+public interface OrchestrationConfig<T extends OrchestrationConfig<T>> {
```

Comment:
(Comment)

![image](https://github.com/user-attachments/assets/3c77679c-10d5-47d1-a89b-1ad83a471b30)

The set of available constructors look slightly confusing/inconsistent to me.

If we wanted to change that, then I would suggest (1) static factory methods. And if I understand correctly there are two distinct creation paths for this class, maybe it makes sense to (2) derive sub-classes for message-based prompts and template-based prompts.

Sub classes of (2) could be returned by factory methods of (1). Thus API remains explorable.